tests/src/JIT/Performance/CodeQuality/BenchI/XposMatrix/XposMatrix.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Xunit.Performance;
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+[MeasureInstructionsRetired(100000)]
+public static class XposMatrix
+{
+    public const int ArraySize = 100;
+
+    static T[][] AllocArray<T>(int n1, int n2) {
+        T[][] a = new T[n1][];
+        for (int i = 0; i < n1; ++i) {
+            a[i] = new T[n2];
+        }
+        return a;
+    }
+
+    static void BoundsCheck<T>(T[][] a, int n1, int n2) {
```

Comment:
I'd like us to not bring the "BoundsCheck" contracts forward. There wasn't a lot of thought put into them initially and I don't really want to set a precedent here.